### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Human018**
Contains Vodafone SMS Service Provider for Laravel Notifications Repository

### CODE REVIEW

File: src/Exceptions/CouldNotSendNotification.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace NotificationChannels\Vodafone\Exceptions;
+
+class CouldNotSendNotification extends \Exception
+{
+    public static function methodDoesNotExist()
+    {
+        return new static(`The toVodafone method does not exist in your notification class.`);
```
Comment (atymic) [inline src/Exceptions/CouldNotSendNotification.php]: backticks should be single quotes (backticks eval code)

File: src/VodafoneChannel.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace NotificationChannels\Vodafone;
+
+use GuzzleHttp\Client;
+use Illuminate\Notifications\Notification;
+use NotificationChannels\Vodafone\Exceptions\CouldNotSendNotification;
+
+class VodafoneChannel
+{
+    /** $var Client */
```
Comment (atymic) [inline src/VodafoneChannel.php]:
```suggestion
    /** $var VodafoneClient */
```

File: src/VodafoneClient.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                'format' => 'json',
+                'flash' => 0,
+            ],
+        ]);
+
+        if (! $res) {
+            throw CouldNotSendNotification::serviceUnknownResponse();
+        }
+
+        $body = json_decode($res->getBody()->getContents())[0];
```
Comment (atymic) [inline src/VodafoneClient.php]:
```suggestion
        $body = json_decode($res->getBody()->getContents())[0] ?? null;
```

Might be worth defaulting to null here

File: src/VodafoneMessage.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        return $this;
+    }
+
+    /**
+     * Set the notification sender.
+     *
+     * @param  string $value
+     * @return $this
+     */
+    public function from($value)
```
Comment (atymic) [inline src/VodafoneMessage.php:50]: Is from required? If so, might be worth adding a check for that (if not, ignore)

Comment (Human018) [reply src/VodafoneMessage.php:50]: 'from' can be left blank, or not included in the request at all and Vodafone will just use a standard number. Only if an additional number or Alphacode is configured in their system would you then include that value in the 'from' field.

Comment (atymic) [reply src/VodafoneMessage.php:50]: Sweet, no worries :)

**Review by Human018 — COMMENTED** [review]
Reviewed and resolved

**Review by atymic — APPROVED** [review]
Looks good to me. Just a few bits from styleCI :)

MWR :)

**Review by Human018 — COMMENTED** [review]
Code styling update

**Review by Human018 — COMMENTED** [review]
Code style update... again

**Pull request merged**